### PR TITLE
✅ Task Scheduling

### LINKED ISSUE #1544: [Idea]: Task Scheduling
### Idea Title

Task Scheduling

### Idea Description

Task Scheduling algorithm using Topological Sorting (Kahn’s Algorithm) implemented in C. This algorithm determines an order in which tasks can be completed given dependencies between them.

### Potential Benefits

- [ ] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [ ] Improves code quality
- [X] Other

### Implementation Suggestions (Optional)

Stack, queues

### PR DESCRIPTION
**Pull request by yashksaini-coder**
### Description

Hi @ajay-dhangar  my PR adds the Task Scheduling using the basic Topological Sorting algorithm, including detailed explanation of the problem, with the solution approach, and the code implementation, also the use of algo & limitations.

Documentation Enhancements:

* Added a new section "Task Scheduling (Topological Sorting)" to `docs/Task Scheduling/Task-schedule.md`, which includes an overview of the task scheduling problem, the solution approach using Kahn’s Algorithm, and a detailed example with code implementation.

Fixes #1544 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by yashksaini-coder** [comment]
@ajay-dhangar can you review my PR

File: docs/Task Scheduling/Task-schedule.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+### Task Scheduling (Topological Sorting)
+
+---
+
+# Task Scheduling Using Topological Sorting in C
+
+This project demonstrates a Task Scheduling algorithm using Topological Sorting (Kahn’s Algorithm) implemented in C. This algorithm determines an order in which tasks can be completed given dependencies between them.
+
+## Table of Contents
```
Comment (ajay-dhangar) [inline docs/Task Scheduling/Task-schedule.md]: ⁉️

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
can you carefully read our contribution guidelines and follow our formats

File: docs/Task Scheduling/Task-schedule.md (hunk 1 above)
Comment (yashksaini-coder) [reply docs/Task Scheduling/Task-schedule.md]: I have added the correct format

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**